whatshap/polyphaseplots.py

```diff
@@ -476,3 +479,764 @@ def draw_threading(
+            [haplotypes[h][pos] for pos in compared_to_phased_pos] for h in range(ploidy)
+        ]
+        (
+            switchflips,
+            switches_in_column,
+            flips_in_column,
+            poswise_config,
+        ) = compute_switch_flips_poly_bt(
+            truth_block,
+            phase_block,
+            report_error_positions=True,
+            switch_cost=1 + 1 / (num_vars * ploidy),
+        )
+
+        # Plot coverage
+        co_parent_ratio = [p / s if s > 0 else 0 for p, s in zip(co_parent_cov, sample_cov)]
+        progeny_ratio = [p / s if s > 0 else 0 for p, s in zip(progeny_cov, sample_cov)]
+        for coverage, color in zip([progeny_ratio, co_parent_ratio], ["tab:purple", "tab:cyan"]):
+            max_coverage = max(coverage)
+            med_coverage = list(sorted(coverage))[len(coverage) // 2]
+            # print(max_coverage)
+            # print(med_coverage)
+            max_coverage = min(max_coverage, 3 * med_coverage)
+            points = [[0, ploidy + y_margin + 2 * coverage[0] / max_coverage]]
+            for pos in range(compared_to_phased_pos[-1]):
+                x1 = x_scale * (pos + x_margin)
+                x2 = x_scale * (pos + 1)
+                points.append([(x1 + x2) / 2, ploidy + y_margin + 2 * coverage[pos] / max_coverage])
+            points.append(
+                [
+                    x_scale * (compared_to_phased_pos[-1] + 1),
+                    ploidy + y_margin + 2 * coverage[-1] / max_coverage,
+                ]
+            )
+            for i in range(0, len(points), 50):
+                point_set = (
+                    [[points[i][0], ploidy + y_margin]]
+                    + points[i : min(i + 51, len(points))]
+                    + [[points[min(i + 50, len(points) - 1)][0], ploidy + y_margin]]
+                )
+                axes.add_patch(
+                    mpatches.Polygon(point_set, color=color, alpha=0.4, closed=True, fill=True)
+                )
+            axes.add_patch(
+                mpatches.Polygon(
+                    [
+                        [0, ploidy + y_margin + 2 * med_coverage / max_coverage],
+                        [
+                            compared_to_phased_pos[-1],
+                            ploidy + y_margin + 2 * med_coverage / max_coverage,
+                        ],
+                    ],
+                    color=color,
+                    closed=False,
+                    fill=False,
+                    ls="-",
+                    lw=0.5,
+                )
+            )
+
+        # Add genome positions
+        try:
+            for pos in range(0, compared_to_phased_pos[-1], 1):
+                x = x_scale * (pos + 1 - 2 * x_margin)
+                axes.text(
+                    x,
+                    ploidy + 2 * y_margin,
+                    phased_positions[pos] + 1,
+                    fontsize=15,
+                    rotation="vertical",
+                )
+        except:
```

Comment:
That’s a bit broad. What is the actual exception you want to ignore?

Reply:
I don't remember. I removed the try-catch and everything seems to work fine.